src/arch/mips/kernel/entry.S

```diff
@@ -21,13 +21,24 @@
  */
  LEAF(mips_second_exception_handler)
 	SAVE_ALL                             /* save all needed registers */
-
+	li $t1, 0x40000000                   /* 0x40000000 is a mask, which mean that Timer Interrupt bit of CAUSE register set in 1 */
+	and  $t1, $k1, $t1                   /* check if Timer Interrupt bit of CAUSE register is 1 */
+	bnez $t1, else                       /* if Timer Interrupt bit of CAUSE register is 1 then go else */
+	nop
+	andi  $k1, $k1, 0x7c                 /* read exception number */
 	PTR_L   $k0, exception_handlers($k1) /* exception number is an offset in array */
 	PTR_LA  $ra, restore_from_exception  /* return address for exit from exception */
 	move    $a0, $sp                     /* Arg 0: saved regs. */
+
 	jr      $k0                          /* Call C code. */
 	nop
+else:
```

Comment:
`else` is not very informative label name. It's better to use something `cp0_timer_irq_hnd:` . Or, if you do not need informative labeling toy can just use `1:`, `2:` etc.

Reply:
done